include/realtime_tools/realtime_box_best_effort.h

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2024, Lennart Nachtigall
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//    * Redistributions of source code must retain the above copyright
+//      notice, this list of conditions and the following disclaimer.
+//
+//    * Redistributions in binary form must reproduce the above copyright
+//      notice, this list of conditions and the following disclaimer in the
+//      documentation and/or other materials provided with the distribution.
+//
+//    * Neither the name of the Willow Garage, Inc. nor the names of its
+//      contributors may be used to endorse or promote products derived from
+//      this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+// Author: Lennart Nachtigall
+
+#ifndef REALTIME_TOOLS__REALTIME_BOX_BEST_EFFORT_H_
+#define REALTIME_TOOLS__REALTIME_BOX_BEST_EFFORT_H_
+
+#include <initializer_list>
+#include <mutex>
+#include <optional>
+#include <utility>
+
+#include <rcpputils/pointer_traits.hpp>
+
+namespace realtime_tools
+{
+
+template <typename T>
+constexpr auto is_ptr_or_smart_ptr = rcpputils::is_pointer<T>::value;
```

Comment:
`is_pointer` trait already checks if it is a poiner or smart pointer. Do we need to redefine it?

Reply:
I guess something like  `is_pointer_v = rcpputils::is_pointer<T>::value` would make more sense. 
The reason I added it in the first place was because I wasn't aware of the rcpputils::is_pointer in the beginning.

Reply:
Just looked it up. It makes sense to call it like that as there is: https://en.cppreference.com/w/cpp/types/is_pointer
Which only checks if a type is a raw pointer

Reply:
Yes, but redefining another trait on top of same trait doesn't make sense right?, I think it make sense to use same `rcpputils::is_pointer<T>::value` without adding another one on top, if they are doing exactly the same job. 

What do you guys think about it? @bmagyar @destogl  @christophfroehlich

Reply:
I have to admit I like it this way as is makes clear that it works with pointers and smart pointers.

Reply:
I don't have a clear opinion about this.